File: Lavcode.Service.Api/Dtos/GetPasswordDto.cs
using Lavcode.Model;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Lavcode.Service.Api.Dtos
{
    public class GetPasswordDto
    {
        public string Id { get; set; }
        public string FolderId { get; set; }
        public string Title { get; set; }
        public string Value { get; set; }
        public string Remark { get; set; }
        public int Order { get; set; }
        public DateTimeOffset UpdatedAt { get; set; }

        public GetIconDto Icon { get; set; }
        public IList<GetKeyValuePairDto> KeyValuePairs { get; set; }

        public PasswordModel ToModel()
        {
            return new PasswordModel()
            {
                Id = Id,
                FolderId = FolderId,
                Title = Title,
                Value = Value,
                Remark = Remark,
                Order = Order,
                UpdatedAt = UpdatedAt,
                Icon = Icon.ToModel(),
                KeyValuePairs = KeyValuePairs.Select(item => item.ToModel()).ToList()
            };
        }
    }
}

File: src/lavcode-node/src/actions/password/dtos/get-password.dto.ts
import { V } from "@halsp/validator";
import { GetIconDto } from "../../../dtos/get-icon.dto";
import { IconEntity } from "../../../entities/icon.entity";
import { KeyValuePairEntity } from "../../../entities/key-value-pair.entity";
import { PasswordEntity } from "../../../entities/password-entity";
import { GetKeyValuePairDto } from "./get-key-value-pair.dto";

export class GetPasswordDto {
  @V().Required()
  id!: string;
  @V().Required()
  folderId!: string;
  @V().IsOptional()
  title?: string;
  @V().IsOptional()
  value?: string;
  @V().IsOptional()
  remark?: string;
  @V().Required()
  order!: string;
  @V().Required()
  updatedAt!: number;

  @V().Required()
  icon?: GetIconDto;
  @V().Required().Items(GetKeyValuePairDto)
  keyValuePairs?: GetKeyValuePairDto[];

  public static fromEntity(
    entity: PasswordEntity,
    icon?: IconEntity,
    keyValuePairs?: KeyValuePairEntity[]
  ) {
    const result = new GetPasswordDto();
    result.id = entity._id;
    result.folderId = entity.folderId;
    result.title = entity.title;
    result.value = entity.value;
    result.remark = entity.remark;
    result.order = entity.order;
    result.updatedAt = entity.updatedAt;
    result.icon = icon ? GetIconDto.fromEntity(icon) : undefined;
    result.keyValuePairs = keyValuePairs?.map((item) =>
      GetKeyValuePairDto.fromEntity(item)
    );
    return result;
  }
}
